utils/build_id.cc

```diff
@@ -41,15 +41,19 @@ static const Elf64_Nhdr* get_nt_build_id(dl_phdr_info* info) {
         p += n->n_descsz;
         p = align_up(p, 4);
     }
-    assert(0 && "no NT_GNU_BUILD_ID note");
+    throw std::runtime_error("no NT_GNU_BUILD_ID note");
 }
 
 static int callback(dl_phdr_info* info, size_t size, void* data) {
     std::string& ret = *(std::string*)data;
     std::ostringstream os;
 
     // The first DSO is always the main program, which has an empty name.
-    assert(strlen(info->dlpi_name) == 0);
+    if (strlen(info->dlpi_name) != 0) {
+        std::string err = "Expected empty DSO name, but got: ";
+        err += info->dlpi_name;
```

Comment:
This looks rather dangerous: you should limit the maximum amount of characters that may be added to the string as a result of this concatenation.